File: library/src/main/java/com/juanimoli/strictmodenotifier/detector/LeakedClosableObjectsDetector.java
package com.juanimoli.strictmodenotifier.detector;

import com.juanimoli.strictmodenotifier.StrictModeLog;

public class LeakedClosableObjectsDetector implements Detector {

    @Override
    public boolean detect(StrictModeLog log) {
        return log.message.contains("A resource was acquired at attached stack trace but never released.");
    }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/detector/LeakedClosableObjectsDetector.kt
package io.github.juanimoli.strictmode.notifier.detector

import io.github.juanimoli.strictmode.notifier.StrictModeLog

class LeakedClosableObjectsDetector : Detector {
    override fun detect(log: StrictModeLog): Boolean {
        return log.message.contains("A resource was acquired at attached stack trace but never released.")
    }
}
